src/mqt/predictor/rl/Predictor.py

```diff
@@ -129,118 +215,20 @@ def evaluate_sample_circuit_using(self, file):
-            str(file).split("/")[-1].split(".")[0].replace("_", " ").split(" ")[-1],
-            qiskit_o3_fid,
-            tket_fid,
-            RL_fid,
-            qiskit_o3_parallel,
-            tket_parallelism,
-            RL_parallelism,
-            qiskit_o3_crit_depth,
-            tket_crit_depth,
-            RL_crit_depth,
-            qiskit_o3_time,
-            tket_time,
-            RL_fid_time,
-            RL_parallelism_time,
-            RL_crit_depth_time,
-            RL_fid_crit_depth,
-            RL_fid_parallelism,
-            RL_crit_depth_fid,
-            RL_crit_depth_parallelism,
-            RL_parallelism_fid,
-            RL_parallelism_crit_depth,
-        )
-
-    def eval_all_sample_circuits(self):
-        res_csv = []
-        res_csv.append(
-            (
-                "Benchmark",
-                "Qubits",
-                "Qiskit O3 Fid",
-                "TKET Fid",
-                "RL Fid",
-                "Qiskit O3 Parallelism",
-                "TKET Parallelism",
-                "RL Parallelism",
-                "Qiskit O3 Crit Depth",
-                "TKET Crit Depth",
-                "RL Crit Depth",
-                "Qiskit O3 Time",
-                "TKET Time",
-                "RL Fid Time",
-                "RL Parallelism Time",
-                "RL Crit Depth Time",
-                "RL Fid Crit Depth",
-                "RL Fid Parallelism",
-                "RL Crit Depth Fid",
-                "RL Crit Depth Parallelism",
-                "RL Parallelism Fid",
-                "RL Parallelism Crit Depth",
-            )
-        )
-
-        results = Parallel(n_jobs=-1, verbose=3, backend="threading")(
-            delayed(self.evaluate_sample_circuit)(file)
-            for file in list(rl.helper.get_path_training_circuits().glob("*.qasm"))
-        )
-        for res in results:
-            res_csv.append(res)
-        # print(results)
-        np.savetxt(
-            rl.helper.get_path_trained_model() / "res.csv",
-            res_csv,
-            delimiter=",",
-            fmt="%s",
-        )
-
-    def instantiate_models(self, timesteps, fid, dep, par, model_name="training"):
-        self.train_all_models(
-            timestep=timesteps,
-            verbose=2,
-            fid=fid,
-            dep=dep,
-            par=par,
-            model_name=model_name,
+        rew_fid = reward.expected_fidelity(transpiled_qc_tket, "ibm_washington")
+        rew_crit_depth = reward.crit_depth(transpiled_qc_tket)
+        rew_gate_ratio = reward.gate_ratio(transpiled_qc_tket)
+        rew_mix = reward.mix(transpiled_qc_tket, "ibm_washington")
+
+        return rl.Result(
+            benchmark=benchmark,
+            used_setup=used_setup,
+            time=duration,
+            fidelity=rew_fid,
+            depth=rew_crit_depth,
+            gate_ratio=rew_gate_ratio,
+            mix=rew_mix,
         )
```

Comment:
as above. this should essentially be just a single line of code.

Reply:
done.